transactions/transactions_service.go

```diff
@@ -226,6 +231,9 @@ func (svc *transactionsService) SendPaymentSync(ctx context.Context, payReq stri
 	}
 
 	err = svc.db.Transaction(func(tx *gorm.DB) error {
+		balanceValidationLock.Lock()
```

Comment:
Is it correct to put this inside the DB transaction rather than outside? e.g. wrap this db transaction in a function with the locking at the beginning of the function?

It would be really great to create a test that definitely fails without the change in this PR (for example do 100 concurrent payments) and confirm this fixes it